adapters/vrtcal/vrtcal.go

```diff
@@ -94,25 +94,37 @@ func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server co
 	return bidder, nil
 }
 
-func getReturnTypeForImp(impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
+func getReturnTypeForImp(mType openrtb2.MarkupType, impID string, imps []openrtb2.Imp) (openrtb_ext.BidType, error) {
 	for _, imp := range imps {
 		if imp.ID == impID {
-			if imp.Banner != nil {
-				return openrtb_ext.BidTypeBanner, nil
-			}
-
-			if imp.Video != nil {
-				return openrtb_ext.BidTypeVideo, nil
-			}
-
-			return "", &errortypes.BadServerResponse{
-				Message: fmt.Sprintf("Unsupported return type for ID: \"%s\"", impID),
+			if mType != 0 {
```

Comment:
This part of code does not have to be in for loop. This function argument is having `mType` as input that is set on seatBid.Bid[i] which can only hold one impression. I would suggest to remove this for loop.

Reply:
Cleaned up this function to reflect your suggestions and updated related test cases.